### ISSUE TITLE
Camera: Please allow access to images without forcing them to be saved first

### ISSUE THREAD

**Issue by OpinionatedGeek** [flutter/flutter#25087]
# TL;DR
I want to be able to access successive pictures from the camera but I don't need to store them. The current Camera plugin mandates that I save the picture to the filesystem, requiring additional permissions.

# Background
I want to be able to scan QR codes. None of the existing libraries are appropriate because all I want is:
* Camera preview in a Widget, not a full-screen activity
* Use of an open-source, transparent decoding API

# Problem
So I thought I'd use the Camera package, take the images from it, pass them to a library like zxing, and be done.

It doesn't really work though. The only way I can see to get images from the Camera package is to call takePicture() which saves it to the filesystem. This is slow, and requires filesystem permissions, as well as being another thing to clean up after exceptions.

At the minute, taking the picture and saving it to the filesystem takes a few seconds on my old dev phone, and while it's doing that the Preview widget freezes, making it really hard to use. (It is working though - loading the file, converting it to the right format and getting zxing to analyse it does detect QR codes.)

All I want, really, is the current picture (repeatedly). (I might even be able to get away with a low resolution version or what appears in the Preview widget.)

Is there a way to access the current picture without writing to the filesystem that I just can't see? Or is there a way to access a stream of pictures from the camera? (Even better if I parameterise it by asking for one picture every quarter of a second, say.)

Or is this entirely the wrong approach, and I shouldn't be passing the image 'up' to Flutter (from the Java code in the Camera package) only to send it 'down' again (to the zxing Java code)?

**Comment by goderbauer** [comment]
@bparrishMines For your ML demo you were piping the camera image directly to the Firebase plugin, right? Any chance that approach would work here as well?

**Comment by chunhunghan** [comment]
I have similar request not to save photo on disk for security reason.
Is it possible that we have another takePicture function without pass filePath and return Uint8List directly.
Thank you for your amazing work.

**Comment by bparrishMines** [comment]
@chunhunghan using `camera.startImageStream()` is currently the only way to get access without saving first. However, you could also save the photo to a temporary file location with the `path_provider` plugin.

**Comment by bparrishMines** [comment]
I'm going to close this issue since the PR has landed. An example project is here: https://github.com/bparrishMines/mlkit_demo

**Comment by rohan20** [comment]
@bparrishMines The example app link now directs to a 404 on your example repo. Could you please provide the correct link?
